pinot-core/src/main/java/org/apache/pinot/core/realtime/converter/RealtimeSegmentConverter.java

```diff
@@ -35,114 +35,116 @@
 import org.apache.pinot.core.segment.creator.impl.SegmentIndexCreationDriverImpl;
 import org.apache.pinot.spi.config.table.ColumnPartitionConfig;
 import org.apache.pinot.spi.config.table.SegmentPartitionConfig;
+import org.apache.pinot.spi.config.table.TableConfig;
 import org.apache.pinot.spi.data.FieldSpec;
 import org.apache.pinot.spi.data.Schema;
 import org.apache.pinot.spi.data.TimeFieldSpec;
 import org.apache.pinot.spi.data.TimeGranularitySpec;
 
 
 public class RealtimeSegmentConverter {
-  private MutableSegmentImpl realtimeSegmentImpl;
-  private String outputPath;
-  private Schema dataSchema;
-  private String tableName;
-  private String timeColumnName;
-  private String segmentName;
-  private String sortedColumn;
-  private List<String> invertedIndexColumns;
-  private List<String> textIndexColumns;
-  private List<String> noDictionaryColumns;
-  private List<String> varLengthDictionaryColumns;
+  private MutableSegmentImpl _realtimeSegmentImpl;
+  private final String _outputPath;
+  private final Schema _dataSchema;
+  private final String _tableName;
+  private final TableConfig _tableConfig;
+  private final String _segmentName;
+  private final String _sortedColumn;
+  private final List<String> _invertedIndexColumns;
+  private final List<String> _textIndexColumns;
+  private final List<String> _noDictionaryColumns;
+  private final List<String> _varLengthDictionaryColumns;
   private final boolean _nullHandlingEnabled;
 
   public RealtimeSegmentConverter(MutableSegmentImpl realtimeSegment, String outputPath, Schema schema,
-      String tableName, String timeColumnName, String segmentName, String sortedColumn,
+      String tableName, TableConfig tableConfig, String segmentName, String sortedColumn,
```

Comment:
remove the  the other unneeded arguments as well (tableName, sortedColumn, inv index columns, text index columnms, etc.). Another PR is also fine

Reply:
All the others are being used to set into the SegmentGeneratorConfig

Reply:
But they can be derived from TableConfig?

Reply:
Oh i see. Maybe they can be. In LLRealtimeDataManager, they're being pulled from IndexLoadingConfig. In HLRealtimeDataManager, they're being used in other places other than RealtimeSegmentConverter. Will see if it can be done in a followup PR